### PR TITLE
Release notes on a single page

### PR DESCRIPTION
**Pull request by nickvigilante**
Fixes the following tickets:

- DOC-2595
- DOC-2598
- DOC-2599

This is the docs repo portion of phase 2 of the release process overhaul.

### CODE REVIEW

File: _redirects [hunk 1]
```diff
@@ -437,6 +437,18 @@ layout: null
 /docs/releases/cockroachcloud-12062021.html /docs/releases/cloud.html 301
 /docs/releases/cockroachcloud-12112020.html /docs/releases/cloud.html 301
 
+#CockroachDB Self-Hosted release notes
+
+{% assign rn_rdr = "v20.2,v21.1,v21.2" | split: "," -%}
```
Comment (ianjevans) [inline _redirects:442]: Nice!

**Comment by ianjevans** [comment]
Coincident with this PR we'll need to update `release-notes.py` to generate the RNs in `_includes/releases`.

**Comment by jseldess** [comment]
> Coincident with this PR we'll need to update `release-notes.py` to generate the RNs in `_includes/releases`.

Yes, as well as our release note generation instructions for writers.

**Comment by nickvigilante** [comment]
@ianjevans @jseldess I've been updating `release-notes.py` in parallel here: https://github.com/cockroachdb/cockroach/pull/76740

File: _redirects (hunk 1 above)
Comment (nickvigilante) [reply _redirects:442]: Thanks! This made redirects much easier. And I tested a few of them and they seem to redirect just fine.

**Review by lnhsingh — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13070)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ianjevans, @jseldess, and @nickvigilante)

___
*[_includes/releases/testing-release-notice.md, line 3 at r3](https://reviewable.io/reviews/cockroachdb/docs/13070#-Mx-knDA4K5k85C0eo41:-Mx-knDA4K5k85C0eo42:b-3o31ze) ([raw file](https://github.com/cockroachdb/docs/blob/11802b8b09348270c517af45b2645fd9f54fe906/_includes/releases/testing-release-notice.md#L3)):*
> ```Markdown
> {% if include.major_version.release_date == "N/A" or include.major_version.release_date < "now" %}{% comment %} check if a release date has been set or if it's in the future {% endcomment %}
> {{site.data.alerts.callout_info}}
> The new features and bug fixes noted on this page are not yet documented across CockroachDB's documentation. Links on this page will direct to documentation for the [latest stable release](../releases/index.html).
> ```

This notice is showing up on v20.2, v21.1, and v21.2 release notes pages - it should only display on v22.1 right?

Also just to clarify, once v22.1.0 is published, a new heading will appear in the sidenav called "Latest v22.1", right? And until then, the alpha/beta/RC will show up in "Latest testing release"?

**Review by nickvigilante — COMMENTED** [review]
> Reviewed 5 of 149 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13070)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ianjevans, @jseldess, and @lnhsingh)

___
*[_includes/releases/testing-release-notice.md, line 3 at r3](https://reviewable.io/reviews/cockroachdb/docs/13070#-Mx-knDA4K5k85C0eo41:-Mx-w7a6752BDAjESahG:b9zrv6c) ([raw file](https://github.com/cockroachdb/docs/blob/11802b8b09348270c517af45b2645fd9f54fe906/_includes/releases/testing-release-notice.md#L3)):*
<details><summary><i>Previously, lnhsingh (Lauren Hirata Singh) wrote…</i></summary><blockquote>

This notice is showing up on v20.2, v21.1, and v21.2 release notes pages - it should only display on v22.1 right?

Also just to clarify, once v22.1.0 is published, a new heading will appear in the sidenav called "Latest v22.1", right? And until then, the alpha/beta/RC will show up in "Latest testing release"?
</blockquote></details>

It should only display on 22.1, correct. I had my lt/gt signs mixed up :laughing: 

The first part, Latest v22.1, will show up. I will try "removing" 22.1 from my local branch as a test and see how the sidebar reacts. Stay tuned.

**Review by lnhsingh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13070)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ianjevans, @jseldess, and @nickvigilante)

___
*[_includes/releases/testing-release-notice.md, line 3 at r3](https://reviewable.io/reviews/cockroachdb/docs/13070#-Mx-knDA4K5k85C0eo41:-Mx-wyUG4rK-3xiHgjA4:b-6pvltu) ([raw file](https://github.com/cockroachdb/docs/blob/11802b8b09348270c517af45b2645fd9f54fe906/_includes/releases/testing-release-notice.md#L3)):*
<details><summary><i>Previously, nickvigilante (Nick Vigilante) wrote…</i></summary><blockquote>

It should only display on 22.1, correct. I had my lt/gt signs mixed up :laughing: 

The first part, Latest v22.1, will show up. I will try "removing" 22.1 from my local branch as a test and see how the sidebar reacts. Stay tuned.
</blockquote></details>

The second part ("until v22.1.0 is published, all alpha/beta/RC shows up in Latest Testing") is what it is doing now, from what I can tell since there's no Latest in v22.1 yet

**Review by lnhsingh — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/13070)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ianjevans, @jseldess, and @nickvigilante)

___
*[_includes/releases/testing-release-notice.md, line 3 at r3](https://reviewable.io/reviews/cockroachdb/docs/13070#-Mx-knDA4K5k85C0eo41:-Mx0G8oGA9rLPg94Hju_:byrwzzf) ([raw file](https://github.com/cockroachdb/docs/blob/11802b8b09348270c517af45b2645fd9f54fe906/_includes/releases/testing-release-notice.md#L3)):*
<details><summary><i>Previously, nickvigilante (Nick Vigilante) wrote…</i></summary><blockquote>

So I simulated "removing" the 22.1 release and the way I had it configured is it removed the "Latest Testing Release" header entirely, seeing as all supported versions were GA at that point. If that behavior is undesirable, let me know and I'll make changes.
</blockquote></details>

I think that makes sense. So once another testing version is released (e.g., v22.2-alpha), the "Latest Testing Release" will show up again, right?

**Review by lnhsingh — APPROVED** [review]

**Review by jseldess — APPROVED** [review]
LGTM!

**Review by ianjevans — APPROVED** [review]

**Pull request merged**